### PR TITLE
NXOS: available_network_resource implementation in nxos_facts

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

##### SUMMARY

resolve: https://github.com/ansible-collections/cisco.nxos/issues/268

##### ISSUE TYPE

 
- Feature Pull Request

##### COMPONENT NAME

nxos_facts
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: tests/integration/targets/nxos_facts/tests/common/all_facts.yaml [hunk 1]
```diff
@@ -21,4 +21,17 @@
...
+  register: result
+  cisco.nxos.nxos_facts:
+    available_network_resources: true
+    gather_network_resources: all
+
+- name: Assert that correct available_network_resources returned
+  assert:
+    that:
+      - result.changed == false
+      - "{{ result['ansible_facts']['ansible_net_gather_network_resources'] | symmetric_difference(result['ansible_facts']['ansible_net_gather_network_resources']) |length\
```
Comment (NilashishC) [inline tests/integration/targets/nxos_facts/tests/common/all_facts.yaml:34]: Are we comparing the same objects here?

Comment (rohitthakur2590) [reply tests/integration/targets/nxos_facts/tests/common/all_facts.yaml]: yeah, that was a mistake, updated. Thanks

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Review by NilashishC — APPROVED** [review]

**Pull request merged**